scripts/stripe/fix-20240926-recorded-duplicated-orders.ts

```diff
@@ -0,0 +1,236 @@
+        SELECT payment_intent_id FROM count_payment_intents WHERE count >= 2
+    ),
+    duplicated_non_paid_orders AS (
+        SELECT 
+            o.id,
+            o.status,
+            duplicated_payment_intents.payment_intent_id,
+            RANK () OVER ( 
+                    PARTITION BY duplicated_payment_intents.payment_intent_id
+                    ORDER BY o."processedAt" NULLS LAST, o."id" ASC
+                ) "duplicated_number",
+            o."createdAt",
+            o."updatedAt",
+            o."SubscriptionId",
+            o."PaymentMethodId",
+            o."description",
+            o."processedAt",
+            o."totalAmount",
+            o."FromCollectiveId",
+            from_collective."slug"
+        FROM "Orders" o
+        INNER JOIN duplicated_payment_intents ON duplicated_payment_intents.payment_intent_id = o."data"#>>'{paymentIntent,id}'
+        JOIN "Collectives" from_collective ON from_collective.id = o."FromCollectiveId"
+        ORDER BY o."data"#>>'{paymentIntent,id}', o."processedAt" NULLS LAST, o."id" ASC
+    )
+    SELECT * FROM duplicated_non_paid_orders WHERE duplicated_number >= 2
+`;
+
+const AFFECTED_TRANSACTIONS_QUERY = `
+    WITH count_payment_intents AS (
+        SELECT o."data"#>>'{previousPaymentIntents,0,id}' AS payment_intent_id, count(1) AS count
+        FROM "Orders" o
+        INNER JOIN "PaymentMethods" pm ON pm.id = o."PaymentMethodId"
+        WHERE TRUE
+        AND pm."service" = 'stripe'
+        AND o.status = 'PAID'
+        AND o."data"#>>'{previousPaymentIntents,0,id}' IS NOT NULL
+        GROUP BY o."data"#>>'{previousPaymentIntents,0,id}'
+    ),
+    duplicated_payment_intents as (
+        SELECT payment_intent_id FROM count_payment_intents WHERE count >= 2
+    ),
+    duplicated_orders AS (
+        SELECT 
+            o.id,
+            duplicated_payment_intents.payment_intent_id,
+            RANK () OVER ( 
+                    PARTITION BY duplicated_payment_intents.payment_intent_id
+                    ORDER BY o."processedAt" ASC
+                ) "duplicated_number",
+            o."createdAt",
+            o."updatedAt",
+            o."SubscriptionId",
+            o."PaymentMethodId",
+            o."description",
+            o."processedAt",
+            o."totalAmount",
+            o."FromCollectiveId",
+            from_collective."slug"
+        FROM "Orders" o
+        INNER JOIN duplicated_payment_intents ON duplicated_payment_intents.payment_intent_id = o."data"#>>'{previousPaymentIntents,0,id}'
+        JOIN "Collectives" from_collective ON from_collective.id = o."FromCollectiveId"
+        ORDER BY o."data"#>>'{previousPaymentIntents,0,id}', o."processedAt" ASC
+    ),
+    orders_to_delete AS (
+        SELECT * FROM duplicated_orders WHERE duplicated_number >= 2
+    )
+    SELECT t.id, t."OrderId", t."createdAt", t."type", t."kind"
+    FROM "Transactions" t
+    INNER JOIN orders_to_delete o ON o.id = t."OrderId" 
```

Comment:
After migrating, we should try and see if any `TransactionSettlements` match these transactions and delete them if that's the case